From d4ad1cebe159d14a52df48a5d0f7371e2198e616 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Sat, 10 Feb 2024 11:04:32 +0800
Subject: [PATCH] clang: add riscv64 support to package script

And drop android and fuchsia build to reduce build time.
---
 tools/clang/scripts/package.py | 62 ++++++++++------------------------
 1 file changed, 18 insertions(+), 44 deletions(-)

diff --git a/tools/clang/scripts/package.py b/tools/clang/scripts/package.py
index 558d7fd2220ea..bebbbaf75c582 100755
--- a/tools/clang/scripts/package.py
+++ b/tools/clang/scripts/package.py
@@ -246,7 +246,7 @@ def main():
     build_cmd = [
         sys.executable,
         os.path.join(THIS_DIR, 'build.py'), '--bootstrap', '--disable-asserts',
-        '--run-tests', '--pgo'
+        '--without-android', '--without-fuchsia', '--pgo'
     ]
     if args.build_mac_arm:
       build_cmd.append('--build-mac-arm')
@@ -305,16 +305,6 @@ def main():
         # Add llvm-readobj (symlinked from llvm-readelf) for extracting SONAMEs.
         'bin/llvm-readobj',
     ])
-    if sys.platform != 'darwin':
-      # The Fuchsia runtimes are only built on non-Mac platforms.
-      want.update([
-          'lib/clang/$V/lib/aarch64-unknown-fuchsia/libclang_rt.builtins.a',
-          'lib/clang/$V/lib/x86_64-unknown-fuchsia/libclang_rt.builtins.a',
-          'lib/clang/$V/lib/x86_64-unknown-fuchsia/libclang_rt.profile.a',
-          'lib/clang/$V/lib/x86_64-unknown-fuchsia/libclang_rt.asan.so',
-          'lib/clang/$V/lib/x86_64-unknown-fuchsia/libclang_rt.asan-preinit.a',
-          'lib/clang/$V/lib/x86_64-unknown-fuchsia/libclang_rt.asan_static.a',
-      ])
   if sys.platform == 'darwin':
     runtime_package_name = 'clang-mac-runtime-library'
     runtime_packages = set([
@@ -358,36 +348,25 @@ def main():
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.asan.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.asan.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.asan.a.syms',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.asan.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.asan.a.syms',
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.asan_static.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.asan_static.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.asan_static.a',
 
         # AddressSanitizer C++ runtime.
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.asan_cxx.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.asan_cxx.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.asan_cxx.a.syms',
-
-        # AddressSanitizer Android runtime.
-        'lib/clang/$V/lib/linux/libclang_rt.asan-aarch64-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.asan-arm-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.asan-i686-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.asan-riscv64-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.asan_static-aarch64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.asan_static-arm-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.asan_static-i686-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.asan_static-riscv64-android.a',
-
-        # Builtins for Android.
-        'lib/clang/$V/lib/linux/libclang_rt.builtins-aarch64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.builtins-arm-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.builtins-i686-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.builtins-x86_64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.builtins-riscv64-android.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.asan_cxx.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.asan_cxx.a.syms',
 
         # Builtins for Linux and Lacros.
         'lib/clang/$V/lib/aarch64-unknown-linux-gnu/libclang_rt.builtins.a',
         'lib/clang/$V/lib/armv7-unknown-linux-gnueabihf/libclang_rt.builtins.a',
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.builtins.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.builtins.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.builtins.a',
 
         # crtstart/crtend for Linux and Lacros.
         'lib/clang/$V/lib/aarch64-unknown-linux-gnu/clang_rt.crtbegin.o',
@@ -396,12 +375,8 @@ def main():
         'lib/clang/$V/lib/armv7-unknown-linux-gnueabihf/clang_rt.crtend.o',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/clang_rt.crtbegin.o',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/clang_rt.crtend.o',
-
-        # HWASAN Android runtime.
-        'lib/clang/$V/lib/linux/libclang_rt.hwasan-aarch64-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.hwasan-preinit-aarch64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.hwasan-riscv64-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.hwasan-preinit-riscv64-android.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/clang_rt.crtbegin.o',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/clang_rt.crtend.o',
 
         # MemorySanitizer C runtime (pure C won't link with *_cxx).
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.msan.a',
@@ -416,34 +391,33 @@ def main():
         'lib/clang/$V/lib/armv7-unknown-linux-gnueabihf/libclang_rt.profile.a',
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.profile.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.profile.a',
-        'lib/clang/$V/lib/linux/libclang_rt.profile-i686-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.profile-x86_64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.profile-aarch64-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.profile-arm-android.a',
-        'lib/clang/$V/lib/linux/libclang_rt.profile-riscv64-android.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.profile.a',
 
         # ThreadSanitizer C runtime (pure C won't link with *_cxx).
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.tsan.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.tsan.a.syms',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.tsan.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.tsan.a.syms',
 
         # ThreadSanitizer C++ runtime.
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.tsan_cxx.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.tsan_cxx.a.syms',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.tsan_cxx.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.tsan_cxx.a.syms',
 
         # UndefinedBehaviorSanitizer C runtime (pure C won't link with *_cxx).
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.ubsan_standalone.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone.a.syms',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.ubsan_standalone.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.ubsan_standalone.a.syms',
 
         # UndefinedBehaviorSanitizer C++ runtime.
         'lib/clang/$V/lib/i386-unknown-linux-gnu/libclang_rt.ubsan_standalone_cxx.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone_cxx.a',
         'lib/clang/$V/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone_cxx.a.syms',
-
-        # UndefinedBehaviorSanitizer Android runtime, needed for CFI.
-        'lib/clang/$V/lib/linux/libclang_rt.ubsan_standalone-aarch64-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.ubsan_standalone-arm-android.so',
-        'lib/clang/$V/lib/linux/libclang_rt.ubsan_standalone-riscv64-android.so',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.ubsan_standalone_cxx.a',
+        'lib/clang/$V/lib/riscv64-unknown-linux-gnu/libclang_rt.ubsan_standalone_cxx.a.syms',
 
         # Ignorelist for MemorySanitizer (used on Linux only).
         'lib/clang/$V/share/msan_*list.txt',
-- 
2.39.2

